src/app/archive/create_schema_new.sql

```diff
@@ -0,0 +1,73 @@
+CREATE TABLE state_hashes
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_state_hashes_value ON state_hashes(value);
+
+CREATE TABLE public_keys
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_public_keys_value ON public_keys(value);
+
+CREATE TABLE snarked_ledger_hashes
+( id    serial PRIMARY KEY
+, value text   NOT NULL UNIQUE
+);
+
+CREATE INDEX idx_snarked_ledger_hashes_value ON snarked_ledger_hashes(value);
+
+CREATE TYPE user_command_type AS ENUM ('payment', 'delegation');
+
+CREATE TABLE user_commands
+( id          serial            PRIMARY KEY
+, type        user_command_type NOT NULL
+, sender_id   int               NOT NULL REFERENCES public_keys(id)
+, receiver_id int               NOT NULL REFERENCES public_keys(id)
+, nonce       bigint            NOT NULL
+, amount      bigint            NOT NULL
+, fee         bigint            NOT NULL
+, memo        text              NOT NULL
+);
+
+CREATE TYPE internal_command_type AS ENUM ('fee_transfer', 'coinbase');
+
+CREATE TABLE internal_commands
+( id          serial                PRIMARY KEY
+, type        internal_command_type NOT NULL
+, receiver_id int                   NOT NULL REFERENCES public_keys(id)
+, fee         bigint                NOT NULL
+);
+
+CREATE TABLE transactions
+( id                  serial PRIMARY KEY
+, hash                text   NOT NULL UNIQUE
+, user_command_id     int    REFERENCES user_commands(id)
+, internal_command_id int    REFERENCES internal_commands(id)
+);
+
+CREATE INDEX idx_transactions_hash ON transactions(hash);
+
+CREATE TABLE blocks
+( id                     serial PRIMARY KEY
+, state_hash_id          int    NOT NULL UNIQUE REFERENCES state_hashes(id)
+, parent_hash_id         int    NOT NULL        REFERENCES state_hashes(id)
```

Comment:
From the perspective of querying chains of blocks, I think it would be better to have this field be `REFERENCES blocks(id)` instead. Also, this field can't be `NOT NULL` because the genesis block will not have a valid reference to a parent.

Reply:
It's possible to let the genesis_block refers to itself. I would prefer to have the parent_id to be `NOT NULL`.

Reply:
I realize that if I made parent_id to be `NOT NULL`, then in order to add a block, it must have its parent block already stored somewhere in the database. That means in order to start an archive serve, we need to download the entire history of the blockchain from somewhere (hopefully another archive node) which is almost impossible.